core/src/services/azblob/core.rs

```diff
@@ -171,34 +171,28 @@ impl AzblobCore {
 }
 
 impl AzblobCore {
+    fn build_path_url(&self, path: &str) -> String {
+        format!(
+            "{}/{}/{}",
+            self.endpoint,
+            self.container,
+            percent_encode_path(&build_abs_path(&self.root, path))
+        )
+    }
+
     pub fn azblob_get_blob_request(
         &self,
         path: &str,
         range: BytesRange,
         args: &OpRead,
     ) -> Result<Request<Buffer>> {
-        let p = build_abs_path(&self.root, path);
+        let mut url = QueryPairsWriter::new(&self.build_path_url(path));
```

Comment:
QueryPairsWriter is optimized for large URIs, so applying it to all URLs doesn't seem like a good idea to me.